demo/src/Components/BottomSheet/BottomSheet.tsx

```diff
@@ -111,23 +214,182 @@ const BottomSheet = (props: HvComponentProps) => {
+              Math.max(
+                -stopPointLocation * SCREEN_HEIGHT - PADDING,
+                -height - PADDING,
+              ),
+            );
+          }
+        }
+      } else if (velocity.value < 0) {
+        // Moving upwards, find the next stop point above
+        for (let i = stopPointLocations.length - 1; i >= 0; i -= 1) {
+          const stopPointLocation = stopPointLocations[i];
+          if (stopPointLocation !== null) {
+            const stopPointY = -SCREEN_HEIGHT * stopPointLocation;
+            if (translateY.value > stopPointY) {
+              nextStopPoint = stopPointLocation;
+            }
+          }
+        }
+        if (nextStopPoint === -1) {
+          const stopPointLocation =
+            stopPointLocations[stopPointLocations.length - 1];
+          if (stopPointLocation !== null) {
+            nextStopPoint = stopPointLocation;
+          }
+        }
+      } else {
+        // Moving downwards, find the next stop point below
+        for (let i = 0; i < stopPointLocations.length; i += 1) {
+          const stopPointLocation = stopPointLocations[i];
+          if (stopPointLocation !== null) {
+            const stopPoint = -SCREEN_HEIGHT * stopPointLocation;
+            if (translateY.value < stopPoint) {
+              nextStopPoint = stopPointLocation;
+            }
+          }
+        }
+        if (nextStopPoint === -1) {
+          const [stopPointLocation] = stopPointLocations;
+          if (stopPointLocation !== null) {
+            nextStopPoint = stopPointLocation;
+          }
+        }
+      }
+
+      if (nextStopPoint > -1) {
+        scrollTo(
+          Math.max(
+            MAX_TRANSLATE_Y,
+            -nextStopPoint * SCREEN_HEIGHT - PADDING,
+            -height - PADDING,
+          ),
+        );
+      }
+    }
+  };
+  const gestureEnabled =
+    hvProps.stopPoints.length > 0 || hvProps.contentSections.length > 0;
+  const gesture = Gesture.Pan()
+    .enabled(gestureEnabled)
+    .onStart(() => {
+      context.value = { startTime: Date.now(), y: translateY.value };
+    })
+    .onUpdate(event => {
+      const currentTime = Date.now();
+      const timeDiff = currentTime - context.value.startTime;
+      const yDiff = event.translationY + context.value.y - translateY.value;
+      velocity.value = yDiff / timeDiff;
+
+      translateY.value = event.translationY + context.value.y;
+      translateY.value = Math.max(translateY.value, MAX_TRANSLATE_Y);
+    })
+    .onEnd(() => {
+      findBottomSheetEndPoint();
+    });
+
+  const overlayStyle = useAnimatedStyle(() => {
+    return { opacity: overlayOpacity.value };
+  });
+
+  const bottomSheetStyle = useAnimatedStyle(() => {
+    return {
+      opacity: contentOpacity.value,
+      transform: [{ translateY: translateY.value }],
+    };
+  });
+
+  const innerView = useMemo(() => {
+    return (
+      <View
+        onLayout={onLayout}
+        onStartShouldSetResponder={() => {
+          console.log(
```

Comment:
let's remove logging or instantiate a logger for debugging (`Logger` in `src/Helpers`)